### PR TITLE
Add verbose flag

### PR DESCRIPTION
**Pull request by SteinHeselmans**
Verbosity needs to summarize the found warnings to the terminal

### CODE REVIEW

File: src/mlx/warnings.py [hunk 1]
```diff
@@ -129,28 +131,47 @@ def check(self, content):
         Args:
             content (str): The content to parse
         '''
-        self.count += len(re.findall(self.pattern, content))
+        matches = re.findall(self.pattern, content, re.M)
+        if self.verbose:
+            for match in matches:
+                print(match)
```
Comment (SteinHeselmans) [inline src/mlx/warnings.py]: this prints the match object and not the original line from the log-file (see run of test cases for details)

**Comment by SteinHeselmans** [comment]
feature is ready for review

note: i am not too sure about the regex of doxygen, hints are welcome

**Comment by Letme** [comment]
I assume regex change is so that we can now print the matched pattern? :+1:  from me (codeclimate needs some fixing thou)

**Comment by bavovanachte** [comment]
FYI: the regex in the Jenkins (Java) plugin:

https://github.com/jenkinsci/warnings-plugin/blob/master/src/main/java/hudson/plugins/warnings/parser/DoxygenParser.java

File: src/mlx/warnings.py [hunk 2]
```diff
@@ -3,25 +3,27 @@
 import re
 import sys
 import abc
-from junitparser import JUnitXml
+from junitparser import JUnitXml, Failure, Error
 import glob
 
-DOXYGEN_WARNING_REGEX = r"(?:(?:((?:[/.]|[A-Za-z]:).+?):(-?\d+):\s*([Ww]arning|[Ee]rror)|<.+>:-?\d+(?::\s*([Ww]arning|[Ee]rror))?): (.+(?:\n(?!\s*(?:[Nn]otice|[Ww]arning|[Ee]rror): )[^/<\n][^:\n][^/\n].+)*)|\s*([Nn]otice|[Ww]arning|[Ee]rror): (.+))$"
+DOXYGEN_WARNING_REGEX = r"(?:((?:[/.]|[A-Za-z]).+?):(-?\d+):\s*([Ww]arning|[Ee]rror)|<.+>:-?\d+(?::\s*([Ww]arning|[Ee]rror))?): (.+(?:(?!\s*(?:[Nn]otice|[Ww]arning|[Ee]rror): )[^/<\n][^:\n][^/\n].+)*)|\s*([Nn]otice|[Ww]arning|[Ee]rror): (.+)\n?"
```
Comment (bavovanachte) [inline src/mlx/warnings.py]: What exactly does this change? The regex seems identical to the one in the jenkins warnings plugin (which is capable of extracting the error message, I think).

Comment (SteinHeselmans) [inline src/mlx/warnings.py]: a small investigation revealed that we must be very lucky to have this regex to match the strings we have. First assumption from jenkins regex explanation `if it is a file we assume doxygen always prints the absolute path` is wrong, and that is where the whole regex is build around. Our test cases and actual doxygen warnings give absolute paths to the files...

i suggest we review the doxygen regex in a seperate issue #38, as that's clearly needed

Comment (SteinHeselmans) [inline src/mlx/warnings.py]: old regex was stripping the filename (not the extension) from the match.group() output

old regex was tuned for line by line input, which is not valid anymore now we throw complete file content to it (test cases didn't cover this multiline, so i added them)

**Comment by bavovanachte** [comment]
:+1:

**Pull request merged**